### PR TITLE
Added mayhem integration

### PR DESCRIPTION
**Pull request by harshavamsi**
Added dockerfile, mayhem file, github actions to integrate mayhem.

### CODE REVIEW

File: fuzz_target_1.mayhemfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+project: unicode-segmentation
```
Comment (xansec) [inline fuzz_target_1.mayhemfile]: Can we rename the fuzz target to something more indicative of what's being fuzzed?

**Review by xansec — CHANGES_REQUESTED** [review]
Please make the requested changes.

**Comment by harshavamsi** [comment]
@xansec Changed the target name as you requested.

File: split_word_bounds.mayhemfile [hunk 2]
```diff
@@ -0,0 +1,5 @@
+project: unicode-segmentation
```
Comment (xansec) [inline split_word_bounds.mayhemfile:1]: I actually was referring to renaming the fuzz target itself. This is fine for now, but check your other submissions and see if you can't rename these.

**Review by xansec — APPROVED** [review]
LGTM!

**Pull request merged**